test/Transition-test.js

```diff
@@ -264,7 +268,7 @@ describe('Transition', () => {
 
         return (
           <Transition
-            ref="transition"
+            ref={transition => this.transition = this.transition || transition}
```

Comment:
@errendir Could you create PR with only ref upgrade in tests? It's unrelated and can be merged faster.

Reply:
OK, I removed the `ref` changes from this PR. I also solved merge conflicts with `master` branch. Please let me know if there is anything else I can do to get this PR merged.

I will create a separate PR for the `ref` changes.